types/shares.go

```diff
@@ -99,7 +99,8 @@ func split(rawData []byte, nid namespace.ID) []NamespacedShare {
 	rawData = rawData[MsgShareSize:]
 	for len(rawData) > 0 {
 		shareSizeOrLen := min(MsgShareSize, len(rawData))
-		rawShare := []byte(append(nid, rawData[:shareSizeOrLen]...))
```

Comment:
Trying to understand exactly what is going on here: this `append(nid, rawData[:shareSizeOrLen]...` is unnecessary  as the share gets namespaced below anyways (`share := NamespacedShare{paddedShare, nid}`)?

Reply:
yep! your other comment is correct, we don't need the copy, only to not use the nid